services/checks/checks.go

```diff
@@ -516,23 +587,34 @@ func (s *Service) processResults(ctx context.Context, check check.Check, target
 	l.Infof("Check script returned %d results.", len(results))
 	l.Debugf("Results: %+v.", results)
 
-	alertsIDs := make([]string, len(results))
+	checkResults := make([]sttCheckResult, len(results))
 	for i, result := range results {
-		id := alertsPrefix + hashID(target.serviceID+"/"+result.Summary)
-		s.createAlert(id, check.Name, target, &result)
-		alertsIDs[i] = id
+		checkResults[i] = sttCheckResult{
+			checkName: check.Name,
+			target:    target,
+			result:    result,
+		}
 	}
 
-	return alertsIDs, nil
+	return checkResults, nil
 }
 
-// non-cryptographic hash
-func hashID(s string) string {
-	data := sha1.Sum([]byte(s)) //nolint:gosec
-	return hex.EncodeToString(data[:])
+// makeID creates an ID for STT check alert.
+func makeID(target *target, result *check.Result) string {
+	s := sha1.New() //nolint:gosec
+	fmt.Fprintf(s, "%s\n", target.serviceID)
+	fmt.Fprintf(s, "%s\n", result.Summary)
+	fmt.Fprintf(s, "%s\n", result.Description)
+	fmt.Fprintf(s, "%v\n", result.Severity)
+	return alertsPrefix + hex.EncodeToString(s.Sum(nil))
 }
 
-func (s *Service) createAlert(id, name string, target target, result *check.Result) {
+type alert struct {
+	alert ammodels.PostableAlert
+	id    string
```

Comment:
We don't actually use that field anywhere anymore (we calculate it and add to labels just below), so that type can be replaced by `ammodels.PostableAlert` directly